.ci/legacy_project/all_packages/android/build.gradle

```diff
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.0'
+        classpath 'com.android.tools.build:gradle:8.1.0'
```

Comment:
@stuartmorgan Do you have any opposition to upgrading the legacy project AGP version here?

The reason the upgrade is required is that the AGP version is shared by the legacy and master versions of the test, and the master version requires AGP 8.1.0.

Reply:
That means we won't notice if we break people on `stable`, which is not ideal. How hard would it be to have the repo tooling adjust this after copying, based on the branch?

Reply:
I can do a simple find and replace somewhere in `create_all_packages_app_command.dart`, perhaps somewhere in https://github.com/flutter/packages/blob/bcb09dbc121e5f886a37cf2808c0d95112276fcd/script/tool/lib/src/create_all_packages_app_command.dart#L130

Is that sort of what you had in mind?

Reply:
Or plumbed through all the way to be part of the args for the command? Something like `--agp-override=7.0.1`?

Reply:
Actually - the legacy project doesn't use the `FlutterExtension` for the compileSdk version. So I can just remove this upgrade